src/hooks/useCurrentExecutionStatus.ts

```diff
@@ -0,0 +1,142 @@
+import { TransactionStatusEnum } from '../types/transactionTypes';
+import getSubstrateDynamicNames from '../util/getSubstrateDynamicNames';
+
+interface Step {
+  chainType: string;
+  label: string;
+  status: string;
+}
+
+export default function useCurrentExecutionStatus() {
+  const [nonceOfTargetFinalizedBlock, setNonceOfTargetFinalizedBlock] = useState<null | number>(null);
+  const [latestReceivedNonceRuntimeApi, setLatestReceivedNonceRuntimeApi] = useState(0);
+  const [steps, setSteps] = useState<Array<Step>>([]);
+  const { dispatchTransaction } = useUpdateTransactionContext();
+  const { api: targetApi } = useApiTargetPromiseContext();
+  const { sourceChain, targetChain } = useSourceTarget();
+  const laneId = useLaneId();
+  const areApiLoading = useLoadingApi();
+  const { currentTransaction } = useTransactionContext();
+  const {
+    bestBlockFinalized,
+    outboundLanes: { latestReceivedNonce: latestReceivedNonceOnSource }
+  } = useDashboard(SOURCE);
+  const { importedHeaders, bestBlockFinalized: bestBlockFinalizedOnTarget } = useDashboard(TARGET);
+  const { latestReceivedNonceMethodName } = getSubstrateDynamicNames(sourceChain);
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+    const getNonceByHash = async () => {
+      const blockHash = await targetApi.rpc.chain.getBlockHash(bestBlockFinalizedOnTarget);
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(
+        latestReceivedNonceMethodName,
+        laneId,
+        blockHash.toJSON()
+      );
+
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      const latestReceivedNonce = latestReceivedNonceCallType.toString();
+      setNonceOfTargetFinalizedBlock(parseInt(latestReceivedNonce));
+    };
+
+    const getLatestReceivedNonce = async () => {
+      const latestReceivedNonceCall = await targetApi.rpc.state.call<Codec>(latestReceivedNonceMethodName, laneId);
+      // @ts-ignore
+      const latestReceivedNonceCallType = targetApi.registry.createType('MessageNonce', latestReceivedNonceCall);
+      setLatestReceivedNonceRuntimeApi(parseInt(latestReceivedNonceCallType.toString()));
+    };
+
+    getNonceByHash();
+    getLatestReceivedNonce();
+  }, [
+    areApiLoading,
+    bestBlockFinalizedOnTarget,
+    currentTransaction,
+    laneId,
+    latestReceivedNonceMethodName,
+    targetApi.registry,
+    targetApi.rpc.chain,
+    targetApi.rpc.state
+  ]);
+
+  useEffect(() => {
+    if (!areApiLoading || !currentTransaction) {
+      return;
+    }
+
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: number | null) => {
+      if (!transactionValue || !chainValue) {
+        return false;
+      }
+      return chainValue > transactionValue;
+    };
+
+    const isDone = (status: boolean | boolean) => (currentTransaction.block && status ? 'DONE' : 'RUNNING');
```

Comment:
`is*` functions should return boolean values, while this one returns a string. I think it's misleading and maybe better to give the function more descriptive name, like `doneStatus` or `completionStatus`?